### PR TITLE
refactor: clean namespaces constants

### PR DESCRIPTION
**Pull request by metacosm**
- refactor!: move SAME_AS_CONTROLLER constant to Constants class, add set
- refactor!: rename WATCH_CURRENT_NAMESPACE_SET to match other constants

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Constants.java [hunk 1]
```diff
@@ -7,13 +7,17 @@ public final class Constants {
 
   public static final Set<String> DEFAULT_NAMESPACES =
       Collections.singleton(Constants.WATCH_ALL_NAMESPACES);
-  public static final Set<String> WATCH_CURRENT_NAMESPACE_SET =
+  public static final Set<String> WATCH_CURRENT_NAMESPACES =
```
Comment (csviri) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Constants.java:10]: Not sure about this, in this simple plural form it says we can watch more current namespace. While the case is that this is only a helper to wrap the WATCH_CURRENT_NAMESPACE. That is why I just added the SET suffix.

Comment (metacosm) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Constants.java:10]: Then let's rename all the Set constants with the `_SET` suffix

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Constants.java:10]: agree!

**Review by csviri — APPROVED** [review]
LGTM

**Pull request merged**